policy-controller/k8s/index/src/http_route.rs

```diff
@@ -34,78 +35,61 @@ impl TryFrom<api::HttpRoute> for InboundRouteBinding {
     type Error = Error;
 
     fn try_from(route: api::HttpRoute) -> Result<Self, Self::Error> {
-        let parents = route
+        let route_ns = route.metadata.namespace.as_deref();
+        let parents = InboundParentRef::convert_all(route_ns, route.spec.inner.parent_refs)?;
+        let hostnames = route
+            .spec
+            .hostnames
+            .into_iter()
+            .flatten()
+            .map(convert::http_match)
+            .collect();
+
+        let rules = route
             .spec
-            .inner
-            .parent_refs
+            .rules
             .into_iter()
             .flatten()
-            .filter_map(
-                |api::ParentReference {
-                     group,
-                     kind,
-                     namespace,
-                     name,
-                     section_name,
-                     port,
-                 }| {
-                    // Ignore parents that are not a Server.
-                    if let Some(g) = group {
-                        if let Some(k) = kind {
-                            if !g.eq_ignore_ascii_case("policy.linkerd.io")
-                                || !k.eq_ignore_ascii_case("server")
-                                || name.is_empty()
-                            {
-                                return None;
-                            }
-                        }
-                    }
-
-                    if namespace.is_some() && namespace != route.metadata.namespace {
-                        return Some(Err(InvalidParentRef::ServerInAnotherNamespace));
-                    }
-                    if port.is_some() {
-                        return Some(Err(InvalidParentRef::SpecifiesPort));
-                    }
-                    if section_name.is_some() {
-                        return Some(Err(InvalidParentRef::SpecifiesSection));
-                    }
-
-                    Some(Ok(InboundParentRef::Server(name)))
-                },
+            .map(
+                |api::HttpRouteRule {
+                     matches, filters, ..
```

Comment:
would it make sense to bind backend_refs to an _ prefixed variable to show explicitly what we're dropping, rather than using `..`?

Reply:
We could. It felt like noise to me, but I'm fine with adding it.

Reply:
agreed, prefer to explicitly match so if the API adds new fields we have to be explicit about handling/ignoring them.

Reply:
done [d06dcb2](https://github.com/linkerd/linkerd2/pull/8949/commits/d06dcb2f67a2db1a8605bbc742b8933ebe2c4ae7)!